classes/class-sensei-media-attachments.php

```diff
@@ -3,30 +3,56 @@
 if ( ! defined( 'ABSPATH' ) ) exit;
 
 class Sensei_Media_Attachments {
-	private $dir;
-	private $file;
+	/**
+	 * The single instance of self.
+	 *
+	 * @var    self
+	 * @access private
+	 * @static
+	 * @since  2.0.0
+	 */
+	private static $instance = null;
+
 	private $assets_dir;
 	private $assets_url;
 	private $token;
 
-	public function __construct( $file ) {
-		$this->file = $file;
-		$this->dir = dirname( $this->file );
-		$this->assets_dir = trailingslashit( $this->dir ) . 'assets';
-		$this->assets_url = esc_url( trailingslashit( plugins_url( '/assets/', $this->file ) ) );
+	/**
+	 * Sensei_Media_Attachments constructor.
+	 *
+	 * @param $file
+	 */
+	private function __construct() {
+		$this->assets_dir = trailingslashit( dirname( SENSEI_MEDIA_ATTACHMENTS_PLUGIN_FILE ) ) . 'assets';
+		$this->assets_url = esc_url( trailingslashit( plugins_url( '/assets/', SENSEI_MEDIA_ATTACHMENTS_PLUGIN_FILE ) ) );
 		$this->token = 'sensei_media_attachments';
 
 		// Localisation
 		$this->load_plugin_textdomain();
 		add_action( 'init', array( $this, 'load_localisation' ), 0 );
```

Comment:
Should this code be moved to 'init' as well?

Reply:
I've put all localization ahead so that our warning messages are localized.